e2e/e2e.py

```diff
@@ -0,0 +1,324 @@
+
+    # AI should have played a second move, I can play again
+    click_button(driver, "#click_1 > rect")
+
+    # Now there should be a piece in #click_1
+    wait_for_presence_of(driver, "#click_1 > circle")
+
+@scenario("two_drivers")
+def can_create_part_and_play(driver1, username1, driver2, username2):
+    """
+    Role: We are two registered users
+    Action: We create and play a full game
+    Result: We see who has won
+    """
+    # Player 1 creates the part
+    click_button(driver1, "#createOnlineGame")
+    select(driver1, "#gameType", "Four in a Row")
+    click_button(driver1, "#launchGame")
+
+    # Player 1 configures the part
+    click_button(driver1, "#firstPlayerCreator") # Player 1 will start
+
+    # Player 2 joins the part
+    click_button(driver2, "#seeGameList")
+    click_button(driver2, "#part_0 > td")
+
+    # Player 1 sees player 2 arrive and selects them
+    click_button(driver1, "#presenceOf_{}".format(username2))
+    click_button(driver1, "#proposeConfig")
+
+    # Player 2 accepts
+    click_button(driver2, "#acceptConfig")
+
+    # Now we are in the game!
+    # Let's play it until the end
+    # First, reload the page to avoid a potential bug (temporary, we can remove this once the bug is fixed)
+    time.sleep(1)
+    driver1.get(driver1.current_url)
+    wait_for_presence_of(driver1, "#playerTurn")
+    click_button(driver1, "#click_3 > rect")
+    wait_for_presence_of(driver2, "#playerTurn")
+    click_button(driver2, "#click_2 > rect")
+    wait_for_presence_of(driver1, "#playerTurn")
+    click_button(driver1, "#click_3 > rect")
+    wait_for_presence_of(driver2, "#playerTurn")
+    click_button(driver2, "#click_2 > rect")
+    wait_for_presence_of(driver1, "#playerTurn")
+    click_button(driver1, "#click_3 > rect")
+    wait_for_presence_of(driver2, "#playerTurn")
+    click_button(driver2, "#click_2 > rect")
+    wait_for_presence_of(driver1, "#playerTurn")
+    click_button(driver1, "#click_3 > rect")
+
+    # Now player 1 has won
+    wait_for_presence_of(driver1, "#youWonIndicator")
+    wait_for_presence_of(driver2, "#youLostIndicator")
+
+def launch_scenarios():
+    """Launches all the scenarios, stop at the first one that fails"""
+    options = Options()
+    options.headless = HEADLESS
+    driver = webdriver.Chrome(options=options)
+    #driver.get("http://localhost:4200")
+
+    for simple_scenario in scenarios["simple"]:
+        # Always go back home for a new scenario
+        driver.get("http://localhost:4200")
+        simple_scenario(driver)
+
+    # Now we need a registered account
+    username = ''.join(random.choices(string.ascii_uppercase + string.digits, k=16))
+    email = username + '@everyboard.org'
+    password = ''.join(random.choices(string.ascii_uppercase + string.digits, k=16))
+
+    register(driver, email, username, password)
+    for registered_scenario in scenarios["registered"]:
+        driver.get("http://localhost:4200")
```

Comment:
Une raison du choix de tester en localhost et pas le faire réellement plus "end to end" en le faisant en develop ?

Reply:
cf discut trello, oui